querybook/server/datasources/comment.py

```diff
@@ -0,0 +1,93 @@
+from flask_login import current_user
+from app.datasource import register
+from logic import comment as logic
+from querybook.server.app.db import DBSession
+from querybook.server.logic.comment_permission import assert_can_edit_and_delete
+
+
+@register(
+    "/comment/data_cell/<int:data_cell_id>/",
+    methods=["GET"],
+)
+def get_comments_by_cell_id(data_cell_id: int):
+    return logic.get_comments_by_data_cell_id(data_cell_id=data_cell_id)
+
+
+@register(
+    "/comment/data_table/<int:data_table_id>/",
+    methods=["GET"],
+)
+def get_comments_by_table_id(data_table_id: int):
+    return logic.get_comments_by_data_table_id(data_table_id=data_table_id)
+
+
+@register(
+    "/comment/data_cell/<int:data_cell_id>/",
+    methods=["POST"],
+)
+def add_comment_to_cell(data_cell_id: int, text):
+    return logic.add_comment_to_data_cell(
+        data_cell_id=data_cell_id, uid=current_user.id, text=text
+    )
+
+
+@register(
+    "/comment/data_table/<int:data_table_id>/",
+    methods=["POST"],
+)
+def add_comment_to_table(data_table_id: int, text):
+    return logic.add_comment_to_data_table(
+        data_table_id=data_table_id,
+        uid=current_user.id,
+        text=text,
+    )
+
+
+@register(
+    "/comment/thread/<int:parent_comment_id>/",
+    methods=["POST"],
+)
+def add_thread_comment(parent_comment_id: int, text):
+    return logic.add_thread_comment(
+        parent_comment_id=parent_comment_id, uid=current_user.id, text=text
+    )
+
+
+@register(
+    "/comment/<int:comment_id>/",
+    methods=["PUT"],
+)
+def edit_comment(comment_id: int, text):
+    with DBSession() as session:
```

Comment:
`with DBSession() as session` is not needed for flask apis, it will always get the same db session for a request.

same below